### ISSUE TITLE
Migration fails with `Cannot create a JSON value from a string with CHARACTER SET 'binary'`

### ISSUE THREAD

**Issue by vogdb** [techouse/sqlite3-to-mysql#115]
**Describe the bug**
Migration fails for JSON values. I'm asking only for advice. I will try to fix it on my own. A starting point would be great. The database I'm trying to migrate is huge, can't paste it here.

**Expected behaviour**
Successful migration.

**Actual result**
Exception

**System Information**

```bash
$ sqlite3mysql --version
```

```
| software               | version                              |
|------------------------|--------------------------------------|
| sqlite3-to-mysql       | 2.1.9                                |
|                        |                                      |
| Operating System       | Linux 5.4.0-177-generic              |
| Python                 | CPython 3.8.10                       |
| MySQL                  | MySQL client not found on the system |
| SQLite                 | 3.31.1                               |
|                        |                                      |
| click                  | 8.1.7                                |
| mysql-connector-python | 8.4.0                                |
| pytimeparse2           | 1.7.1                                |
| simplejson             | 3.19.2                               |
| tabulate               | 0.9.0                                |
| tqdm                   | 4.66.2                               |
```

**Additional context**
MySQL is via Docker

In case of errors please run the same command with `--debug`. This option is only available on v1.4.12 or greater.
```
2024-05-01 17:08:02 ERROR    MySQL transfer failed inserting data into table directus_collections: 3144 (22032): Cannot create a JSON value from a string with CHARACTER SET 'binary'.
Traceback (most recent call last):
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/bin/sqlite3mysql", line 8, in <module>
    sys.exit(cli())
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/sqlite3_to_mysql/cli.py", line 165, in cli
    SQLite3toMySQL(
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/sqlite3_to_mysql/transporter.py", line 731, in transfer
    self._transfer_table_data(sql=sql, total_records=total_records)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/sqlite3_to_mysql/transporter.py", line 635, in _transfer_table_data
    self._mysql_cur.executemany(
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/mysql/connector/cursor.py", line 1535, in executemany
    self.execute(operation, params)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/mysql/connector/cursor.py", line 1510, in execute
    res = self._connection.cmd_stmt_execute(
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/mysql/connector/opentelemetry/context_propagation.py", line 102, in wrapper
    return method(cnx, *args, **kwargs)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/mysql/connector/connection.py", line 1464, in cmd_stmt_execute
    result = self._handle_binary_result(packet)
  File "/home/vogdb/workspace/friendlee/x5fond/migrate/venv/lib/python3.8/site-packages/mysql/connector/connection.py", line 1354, in _handle_binary_result
    raise get_exception(packet)
mysql.connector.errors.DataError: 3144 (22032): Cannot create a JSON value from a string with CHARACTER SET 'binary'.
```

**Comment by techouse** [comment]
Hey,

Not sure what else I can do here except point you to this StackOverflow issue https://stackoverflow.com/questions/38078119/mysql-5-7-12-import-cannot-create-a-json-value-from-a-string-with-character-set

**Comment by vogdb** [comment]
Thank you! I have MySQL 8.0.36. I solved the problem by using 8.0.35. Closing.
